mcda/utils/utils_for_main.py

```diff
@@ -429,14 +473,22 @@ def save_df(df: pd.DataFrame, folder_path: str, filename: str):
     timestamp = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
     new_filename = f"{timestamp}_{filename}"
 
-    result_path = os.path.join(folder_path, new_filename)
-    df.to_csv(path_or_buf=result_path, index=False)
+    full_output_path = os.path.join(output_directory_path, folder_path, new_filename)
+    try:
+        ensure_directory_exists(os.path.dirname(full_output_path))
+    except IOError:
```

Comment:
Same comment as above, here as well exception will not be caught due to base Exception already handled.